homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +88,59 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_HaServiceInfoDescription = TypedDict(
+    "_HaServiceInfoDescription",
+    {
+        "x-homeassistant-matching-domains": set,
```

Comment:
Let's call this `x_homeassistant_matching_domains` to use normal class syntax, and update `ATTR_HA_MATCHING_DOMAINS` to match.

Reply:
I think this might be considered a breaking change for custom components.
What is the view of the core team on this one?

Reply:
I have adjusted the definition for `_HaServiceInfoDescription`, and added a Breaking Change section at the top to warn in case the string is used by custom components.
It also allows us to import the standard `TypedDict` from `typing` without relying on `typing_extensions`.